### ISSUE TITLE
`-nologo` should be the default

### ISSUE THREAD

**Issue by terrajobst** [PowerShell/PowerShell#15644]
## Steps to reproduce

Launch PowerShell in your favorite terminal.

## Expected behavior

The prompt I've set up in my PowerShell profile.

## Actual behavior

I get the following boilerplate:

```none
PowerShell 7.1.3
Copyright (c) Microsoft Corporation.

https://aka.ms/powershell
Type 'help' to get help.
```

"But Immo, you can turn this off by using `-nologo`". Yes. But I have to do this in every single one of the places that launches the shell: shortcuts in the start menu, pinned task bar entries, the Windows Terminal commands, the VS Code integrated terminal, the Visual Studio terminal, Git GUI clients etc. It's tiring. If there would be a user wide setting, that might be an OK workaround, but every time I set up a new machine I have to add these extra arguments -- just to get a shell that is ready for use.

The year is 2021. We're embracing command line stuff and simple text files for the modern dev environment (really, we're just rediscovering Unix philosophy, but that's a different rant 😄).  It seems the default shell behavior should optimize for this kind of user, rather than forcing this boilerplate on everyone that arguably adds zero value to their experience.

**Comment by bergmeister** [comment]
The point around configuration in multiple places is a good one. Being able to override the default (whatever it is now or will be) via an environment variable seems like a good addition.

**Comment by kilasuit** [comment]
I do like this idea but don't think this should be an environmental variable but should be instead added as an option in the powershell.config.json file as documented at https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_powershell_config?view=powershell-7.1

**Comment by warrenrumak** [comment]
I'm generally in favour of this but the question must be asked -- without any kind of heading, how would a user know whether they're in the CMD or PowerShell environment?  (The existence of a space between the > and the cursor is not a satisfactory answer.)

**Comment by terrajobst** [comment]
I don’t believe this to be a general concern as virtually all places show this information contextually (title bar, tab icon etc). In the rare cases where a user might not know, entering commands will make it quite obvious.

**Comment by rkeithhill** [comment]
I'm not a big fan of all the text but it's not completely without precedent:
```
2> python
Python 2.7.9 (default, Dec 10 2014, 12:28:03) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>>
```
At least the above is only two lines of extra text.  :-)  I wouldn't mind seeing just:
```
PowerShell 7.1.3
PS C:\Users\hillr> 
```
Or if your legal team absolutely requires the copyright:
```

**Comment by rkeithhill** [comment]
Argh GH!!  There is some keyboard shortcut I keep hitting that presses the `Close with comment` button.  Sorry guys.

**Comment by Nirmal4G** [comment]
Maybe, if there's a way to check the launch context of the console, we could detect whether it's lauched by the user (_interactive_) or via script (_non-interactive_). This way we can display the header for the first-time the user launches the console and have a setting that suppress the header afterwards, while suppressing the header in the context of a script launch.

**Comment by iSazonov** [comment]
More than once we have received requests to change default options. We already have several new environment variables (update notifications, telemetry). This is a good way, we just have to be more consistent to have support for these options in all kind of configurations (environment variables, config files, Registry)

**Comment by SteveL-MSFT** [comment]
I think having some cmdlets making it easy to configure powershell.config.json might be the way to do this (since this file is already being read at startup).  I'd hate to add more env vars.

**Comment by iSazonov** [comment]
> I'd hate to add more env vars.

We need to have a consistency with environment variables, config files, Registry. In Enterprise it is better use GPO and Registry and it is huge headache to use config files. In DevOps it is better to use environment variables, In personal environment it is better to use config files.

**Comment by JustinGrote** [comment]
> I think having some cmdlets making it easy to configure powershell.config.json might be the way to do this (since this file is already being read at startup).  I'd hate to add more env vars.

Would there be a userspace way to do this? Pretty sure that's a protected folder.

**Comment by kilasuit** [comment]
@JustinGrote  - there is as per the doc & below
![image](https://user-images.githubusercontent.com/6355225/123065528-2117ce80-d407-11eb-9a83-3e3c1a049494.png)

**Comment by dazzknowles** [comment]
Agree they should be off by default but the one time they are useful is to make the user aware of a new version.

**Comment by iSazonov** [comment]
You can put `cls` in your user profile :-)

**Comment by SeeminglyScience** [comment]
> You can put `cls` in your user profile :-)

That's what I do, but it's worth noting that that is pretty annoying for nested pwsh instances.

**Comment by SteveL-MSFT** [comment]
@iSazonov I agree we have a bit of a mess right now with env vars and config.  As for the registry, the reason we created the config file was to move al config out of the registry.

**Comment by iSazonov** [comment]
> the reason we created the config file was to move al config out of the registry.

Config files are terribly inconvenient for administrators who manage hundreds and thousands of servers and workstations in Enterprises. GPO is more fast and flexible.

**Comment by kilasuit** [comment]
@iSazonov you can deploy a config file that you've kept in source control at scale easily enough using GPO already & I'd say that would be cleaner than the GPO UI

**Comment by iSazonov** [comment]
> @iSazonov you can deploy a config file that you've kept in source control at scale easily enough using GPO already & I'd say that would be cleaner & much more efficent than navigating the GPO UI for the configuration properties

There are undoubtedly many applications that allow you to manage configurations. But I prefer GPOs and sleep well at night instead of getting emergency alerts. :-) I guess I'm not alone in disliking round-the-clock alarms. :-)

**Comment by thedavecarroll** [comment]
> @iSazonov I agree we have a bit of a mess right now with env vars and config. As for the registry, the reason we created the config file was to move al config out of the registry.

Doesn't cross-compatibility preclude the use of the registry? Why not just focus on a single method or on those methods available to all supported operating systems, like config files and environment variables?

**Comment by iSazonov** [comment]
> Doesn't cross-compatibility preclude the use of the registry?

If I'm talking about GPOs and Registry, am I talking about Unix? 
I'm talking about the fact that I manage an environment of hundreds and thousands of Windows servers and workstations and configuring with GPOs is the most powerful and convenient tool - it works out of the box and I don't need to spend time and money on third party applications. 
Since you mentioned cross-compatibility you might want to take a look at Mono's implementation of Registry.  :-)

**Comment by kilasuit** [comment]
Still can do so with GPO just get it to use this instead of a painful UI @iSazonov https://www.adamfowlerit.com/2016/07/group-policy-preferences-replace-existing-file/

**Comment by Rexogamer** [comment]
> Or if your legal team **absolutely requires** the copyright:
> 
> ```
> PowerShell 7.1.3 - Copyright (c) Microsoft Corporation
> PS C:\Users\hillr>
> ```
Personally, I'm a fan of this - seeing the version on start is useful (certainly for preview builds) and the copyright text doesn't feel *that* intrusive, but it cuts down the bloat quite nicely

**Comment by theJasonHelmick** [comment]
Hi @terrajobst 

I agree with @bergmeister that your point  around configuration in multiple places is a problem. I had the same initial reaction you described “Why Immo, just change…” - or - “Well, don’t close your shell!” ;) But you raise a very valid point.

If we were to take up this issue, which probably wouldn’t happen until after 7.2 LTS release, do you agree with the suggestions below that would reduce the foot print of the logo, and make it possible to turn off in powershell.config.json?

1. @rkeithhill  - Reduced footprint of logo 
```output
PowerShell 7.1.3 - Copyright (c) Microsoft Corporation
PS C:\Users\MyName>
```
2. @SteveL-MSFT  - Global setting in powershell.config.json

What does everyone think?

**Comment by terrajobst** [comment]
> powershell.config.json

Where does this file live? Is this file for a specific version of PowerShell (i.e. do I have to re-apply when I move to 7.3 for example)? Does it roam across machines? Assuming it's a roaming file and it's applicable to all PowerShell sessions irrespective of the command line that started it, that would work for me.

> do you agree with the suggestions below that would reduce the foot print of the logo

At least for me any logo, even the reduced one is noise that I'd like to turn off. Since I'll turn it of no matter what the reduction won't do much for me.

**Comment by theJasonHelmick** [comment]
@terrajobst Get-Help about_powershell_config has this information, including the location of the file in $PSHOME - and I think would help in most situations.

**Comment by terrajobst** [comment]
> A powershell.config.json file in the $PSHOME directory defines the configuration for all PowerShell Core sessions running from that PowerShell Core installation.

That's not inspiring confidence as that means I need to babysit this for every new version on every machine.

**Comment by SeeminglyScience** [comment]
There's also a user scoped location in `SpecialFolders.Personal + "\PowerShell"` on Windows.  That one is more what you're looking for.

**Comment by kilasuit** [comment]
@terrajobst I just raised an issue that should help to combat this ` I need to babysit this for every new version on every machine` for the file that is stored in the install directory & I think right now in-place upgrades may override it.

regarding your other questions
>Where does this file live? 

in many places, including in the install directory by default or you can also have one in your user directory as per the doc linked above

> Is this file for a specific version of PowerShell (i.e. do I have to re-apply when I move to 7.3 for example)? 

Technically yes, though more for installation method than version (other than stable/preview)
I think you may need to currently reapply on upgrade installs, only in the install location (All Users) however I don't want that either in future as this file could potentially have some settings applied using GPO's & raised #15796 for this.

> Does it roam across machines? 

If you place your settings in a `powershell.config.json` file within your user profile then yes it would if you use a roaming profile

> Assuming it's a roaming file and it's applicable to all PowerShell sessions irrespective of the command line that started it, that would work for me.

If you make use of the user profile location for your required settings this should then apply to all versions of both `pwsh` & `pwsh-preview` regardless of install method MSI / MSIX / ZIP / MS Store / dotnet global tool / etc
or at least that is how I understand it to be & **should** meet your needs.

**Comment by theJasonHelmick** [comment]
I'll add to the conversation that we are checking with legal to see if we could trim the logo to something similar to:
```output
PowerShell 7.2
PS>
```

@kilasuit outlined some good points.  I think if we could agree on adding a setting to powershell.config.json to completely suppress the logo, maybe someone has thoughts on tooling (cmdlet support) to discover and manage those settings without directly editing the json.

**Comment by JustinGrote** [comment]
@theJasonHelmick as long as the setting can be done in userspace I'm cool with that.

**Comment by dancojocaru2000** [comment]
I would really like this in order to replace the default banner with a custom one.

**Comment by m1karii** [comment]
> maybe someone has thoughts on tooling (cmdlet support) to discover and manage those settings without directly editing the json.

Yeah. I'd prefer being able to change this setting from within the shell itself.

**Comment by theJasonHelmick** [comment]
I'm looking at a minimum viable for this enhancement to get us started on improving the display of the logo, and including a machine wide configuration to enable/disable the display of the logo.  I appreciate @terrajobst  suggestions and will continue to investigate the value of of multi-machine configuration  other than GPO.

To move forward, here’s what I’m thinking:

* New minimal terminal logo similar too:
```powershell
PowerShell 7.1.3 
PS C:\Users\MyName>
```
* Add additional configuration setting to `powershell.config.json` in **$PSHome** following the same system wide and current user options discussed in `About_PowerShell_Config`.
* Perhaps something like:  “PowershellTeminalLogo”:  “True”

Tooling support isn’t nessisary for the initial implementation, but could be added:
```powershell
Get-PowerShellConfigurationSetting
# Lists all settings
Get-PowerShellConfigurationSetting -Name “PowerShellTerminalLogo”
# List only the specified setting
Set-PowershellConfigurationSetting -Name “PowerShellTerminalLogo -Value True
# Sets the setting value
Get-PowerShellConfigurationSetting -Name “PowerShellTerminalLogo” | Set-PowerShellConfiguratinSetting -Value True
```

**Comment by blowdart** [comment]
If nologo tends towards the default I'd suggest (and want for myself) a slightly different nologo. The "problem" I have with nologo is it suppresses the update available message. Given that the update may be a security update I'd like to have a -nologoexceptforupdates (yes, msft employees are bad at naming) option which suppresses startup messages _except_ for the update message.

**Comment by kilasuit** [comment]
@blowdart I agree though perhaps `-updatelogoonly` or just simply `-updatelogo` is a better name for this

**Comment by blowdart** [comment]
This is why no one lets me name things 😀

**Comment by rkeithhill** [comment]
> the "problem" I have with nologo is it suppresses the update available message

This is why I don't think `-nologo` should be the default.  I want a "reduced" logo  that is the default.  There is still a place for `-nologo`.  Also, the PS update notification in the prompt already has a separate mechanism to control it -  `$env:POWERSHELL_UPDATECHECK`.

So for `-nologo`, I want it to behave exactly like it does today. 
```
PS C:\Users\hillr>
```
I'd prefer to see a "default" startup experience like this:
```
PowerShell 7.3.0
PS C:\Users\Keith>
```
And for the default / "reduced" logo, if there happens to be a PS update available AND I have not disabled the update check, I would expect to see this
```
PowerShell 7.3.0

   A new PowerShell stable release is available: v7.3.1
   Upgrade now, or check out the release page at:
     https://aka.ms/PowerShell-Release?tag=v7.3.1

PS C:\Users\hillr> 
```

That leaves just the "where/how to get help" text: 
```
https://aka.ms/powershell
Type 'help' to get help.
```
Which is great for the **first time** user but this text quickly becomes extra noise on every session start.  Noise because you have trained me with the rolled up newspaper, er this text over hundreds and hundreds of new PS session starts.  I really, I mean REALLY don't need to see it anymore.  😜   At the very least, just put the `https://aka.ms/powershell` link in the help topic for `help`.  You could then simplify to:
```
PowerShell 7.3.0 (type 'help' to get help)
PS /home/hillr>
```
That is **if** the team deems it necessary to always display info on how to get help.

**Comment by iSazonov** [comment]
1. In future we will have pwsh in box so we should have consistency with cmd.exe and Windows PowerShell.
2. Did anybody ask to remove Python logo? 😄 @terrajobst you use Python in .Net repository, did you request them too? Link?
3. We support side-by-side and global tool. So it is important to see what current pwsh distribution/version is. I think it makes sense to have _specific logo for every kind of pwsh_ - personal(application)-wide, system-wide, global tool - not only version. We never discussed this.
Ex., VS Code run pwsh - system-wide, installed with VS Code, from my profile? can I have specific config for the scenario? Or whether  I run pwsh from Global tool and with specific config? Should we support host specific configs again?
4. Add more options to configs/envs is bad idea. It is difficult to discover and difficult to replicate. Users can put `cls` to user profile. On Windows the profile can be synced with OneDrive.
5. We want to keep update notification.


We need PowerShell Committee conclusion before change something, if change.

**Comment by rkeithhill** [comment]
> have specific logo for every kind of pwsh

🤔 I'm not sure this user would like that, but I think it would be useful info to have in `$PSVersionTable` - particularly for support scenarios.  IOW I don't think users in general should care about which one they're using but if they run into trouble and we're trying to help them, it would be good if they could easily tell us (report in an issue) which kind of install of PowerShell they have.

**Comment by iSazonov** [comment]
>  I'm not sure this user would like that, 

Follow looks bad?
```
PowerShell 7.3.0 (Global tool)
PS C:\Users\Keith>
```
```
PowerShell 7.3.0 (System Wide)
PS C:\Users\Keith>
```

**Comment by rkeithhill** [comment]
It's not that it "looks bad".  The concern is whether it would cause concern/confusion to users who don't care or know what System Wide vs Global tool even means.  

That said, if we went the Python route and displayed multiple attributes in parens then it might not be as confusing i.e. I think the more "info" in the parens, the more likely folks will TL;DR. 😄   For example:

```
PowerShell 7.3.0 (system, x64, Jun 7 2022, type 'help' to get help)
PS C:\Users\Keith>
```
And there's also the issue of what other values are there besides `system` and `global tool`?  `snap`, `store`, ??

**Comment by kilasuit** [comment]
I semi like that @iSazonov but also agree with @rkeithhill that info should also be in `$PSVersionTable`

**Comment by iSazonov** [comment]
> The concern is whether it would cause confusion to users who don't know or care what System Wide vs Global tool even means.

There is an important conception - PowerShell host. Every PS host can have custom config. I think this is important to point out whether an user understands these concepts or not.

> And there's also the issue of what other values are there besides `system` and `global tool`? `snap`, `store`, ??

Logon shell on Unix.

**Comment by rkeithhill** [comment]
> Every PS host can have custom config

@iSazonov  Do you really mean `host` or are you referring to different types of PS installation?  Because different hosts can pick their own startup banners - like PSES does in Visual Studio Code:

![image](https://user-images.githubusercontent.com/5177512/143792052-3327d585-1644-497b-8ddb-59b78ee2098f.png)

About twelve years ago, I wrote a `Make-PS1ExeWrapper.ps1` script that created a PS host to run a script extracted from an embedded resource in the compiled exe.  That host, for obvious reasons, did not print any startup banner.

**Comment by iSazonov** [comment]
> Do you really mean host or are you referring to different types of PS installation?

I generalize. I think it is useful to see which pwsh is actually running - in logs, in traces, in a remote session and in the console too. It doesn't make sense to distinguish between host apps for system-wide and user-wide installations, but for global tools it does.

**Comment by SteveL-MSFT** [comment]
@PowerShell/powershell-committee discussed this.  We looked at other shells like `zsh`, `csh`, `bash` and each of them simply show a prompt without additional information as the user explicitly started those shells and should know what they are using.  We agreed to have the default not show any logo information.  However, upgrade information will still be shown unless suppressed by existing means.  A new `-Logo` switch will be added to retain current behavior (or just `PowerShell <version>`).  Waiting on legal and marketing approval before finalizing.

**Comment by daxian-dbw** [comment]
@SteveL-MSFT I think having `version` information at startup is useful, especially given that many users uses both stable and preview versions of PowerShell at the same time.

**Comment by SteveL-MSFT** [comment]
@daxian-dbw perhaps preview version should default to `-logo` with simplified `PowerShell <version>`?

**Comment by iSazonov** [comment]
> @SteveL-MSFT I think having `version` information at startup is useful, especially given that many users uses both stable and preview versions of PowerShell at the same time.

It is true for global tools, VS Code and others.
Moreover, we will most likely be asked to implement a separate configuration for such scenarios, host applications.
Perhaps they (every) should be specific host application and this will be the correct answer to this issue?

**Comment by dancojocaru2000** [comment]
Looking through this issue, I'm not sure if this is considered: is there a way for the user to change the "logo" in the same way the prompt can be changed?

`fish`, for example, allows the user to define a `fish_greeting` function. If that function exists, `fish` doesn't output anything anymore, and instead calls the function, and the function should output the greeting.

If such a feature would be implemented in PowerShell, disabling the logo would be as simple as defining an empty function.

**Comment by JustinGrote** [comment]
@dancojocaru2000 a problem with that approach is currently the logo is emitted well before the runspace is available to keep the appearance of being "snappy", it might feel sluggish if implemented that way.

**Comment by dancojocaru2000** [comment]
@JustinGrote The only thing a user wants to do upon starting PowerShell is run a command, which a user cannot do until the runspace is available. As such, how quick the first text appears is rather irrelevant, considering the prompt or the characters typed don't appear.

**Comment by brwilkinson** [comment]
commenting to follow... 

Also, this is what I always configure in all of my shortcuts 
- nologo on startup argument
- profile has some message of what ps version I am in and which profile loaded.

### pwsh
![image](https://user-images.githubusercontent.com/3605266/145306195-094a48d8-ff9a-49d6-865d-a78d45693485.png)

### powershell
- I assume I will still need this customization, since this thread is just for pwsh

![image](https://user-images.githubusercontent.com/3605266/145306245-5d2c03d6-a3d3-48c2-912e-c09a9bdc2be1.png)

#### Terminal configuration

![image](https://user-images.githubusercontent.com/3605266/145306276-59cc9d60-6f06-4584-a16f-7ae90e558e59.png)

#### vscode configuration

![image](https://user-images.githubusercontent.com/3605266/145306387-44ea49d4-b55f-43c5-98b3-50a575a23ca0.png)

So if i can achieve this with less custom settings, I would be fine with that change.

**Comment by rkeithhill** [comment]
> a problem with that approach is currently the logo is emitted well before the runspace is available to keep the appearance of being "snappy", it might feel sluggish if implemented that way.

I agree. I have my Windows Terminal profile for PWSH set to use `-nologo` and my profile takes about 2-3 seconds to load.  It is a bit disconcerting to see a blank console for that long.

**Comment by rkitover** [comment]
I had an idea, we already have `$env:POWERSHELL_UPDATECHECK`, we could also add `$env:POWERSHELL_LOGO` or perhaps `$env:POWERSHELL_SILENT` for both or some such.

**Comment by SteveL-MSFT** [comment]
@PowerShell/powershell-committee discussed this again and we agreed to default to:

```
PowerShell <version>
```

For the startup logo.  `-Nologo` retains its current behavior.  In the case of preview, we would expect:

```
PowerShell 7.3-preview.2
```

as an example.

**Comment by iSazonov** [comment]
@SteveL-MSFT Did you discuss https://github.com/PowerShell/PowerShell/issues/15644#issuecomment-981244436?
It seems we need to take back ShellId in Engine (see DefaultPowerShellShellID in code).

**Comment by kilasuit** [comment]
I've been thinking that on first ever execution of pwsh (inc new versions) the current logo (or any updates per version inc a link to the changelog) should be ok, but every other execution the slim one as per @SteveL-MSFT comment above would be desirable

any thoughts on that as an approach?

**Comment by iSazonov** [comment]
> any thoughts on that as an approach?

This is always easy to do on Windows using Registry but the PowerShell team prefers config file and this causes unreasonable startup delay.

**Comment by dancojocaru2000** [comment]
Well, it wasn't exactly fixed. The logo is smaller, but still there. 

I'm not sure yet if there is an option to disable the logo when passing command line arguments is not possible.

**Comment by terrajobst** [comment]
Agreed. Until there is a user wide setting to disable the logo entirely I personally don’t think my issue was addressed.

**Comment by rkitover** [comment]
What about my idea of having an environment variable for this? 

There is already `$env:POWERSHELL_UPDATECHECK` which is definitely nice to have.

Maybe we don't want to have 20 environment variables, and maybe we don't want to have `$env:POWERSHELL_OPTIONS` for security reasons, but this seems like a good solution for this particular problem.

On Linux this would allow putting this environment variable in pam or systemd environment definitions and use pwsh as a login shell with no options and no startup banner.

On Windows it would allow setting one machine env var and not have to tweak a bunch of different things to pass `-nologo`.

**Comment by kilasuit** [comment]
@dancojocaru2000 / @terrajobst - I didn't say I agreed with the close resolution (as I don't) b

whilst env vars can work, personally I dislike them and I'd like to see them and [this implemented too](https://github.com/PowerShell/PowerShell/issues/15796)

**Comment by m1karii** [comment]
@rkitover agreed. At least something like `$env:POWERSHELL_DISPLAYLOGO` would be nice, if nothing else.

**Comment by SeeminglyScience** [comment]
Re-opening as I believe this was closed by mistake.

**Comment by jeremyj563** [comment]
I don't really care that much about which is the default behavior but I definitely agree there should be a global flag of some sort for this. I'd prefer the environment variable approach as I would always know exactly where to look/set the config regardless of host OS.

**Comment by SteveL-MSFT** [comment]
The default logo has been minimized to just show the version now.  I don't think additional work here is needed.

**Comment by JustinGrote** [comment]
Sorry to necro this but if you really want a minimal terminal if you are setting PowerShell as your default linux login, you can do these steps:
1. Create a file /usr/local/bin/pwshlogin, make it executable (`chmod +x /usr/local/bin/pwshlogin`), and add the following content:
```
#!/bin/sh
pwsh -nologo -l
```

2. `usermod -s /usr/local/bin/pwshlogin YOURUSERNAME`

3. BEFORE YOU CLOSE THE WINDOW, Test with a new ssh login or otherwise to make sure it works.
4. Voila, clean login:
![image](https://github.com/PowerShell/PowerShell/assets/15258962/ac056278-2433-47b3-81e3-82d1e8b4313c)

**Comment by rdebath** [comment]
@JustinGrote 
1) Your script is wrong, it leaves a bourne shell running.
2) $SHELL is now broken if you manage to make it alogin shell
3) Most systems allow only root to define shells not in a whitelist.
4) Using another program to fix inadequacies of your shell should be a serious embarrassment.

**Comment by m1karii** [comment]
still waiting
